### DISCUSSION TITLE
assignment 7th 관련 질문

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#84] [Q&A] [answered]
malloc/free와 new/delete의 차이 에서
new operator는 사용자가 오버로딩한 함수고
operator new는 그냥 함수 인건가요?

**Comment by Woon-2** [comment, marked as answer]
간단합니다.   
`new operator`를 `overload`하셨나요, `operator new`를 `overload`하셨나요.

우리는 `operator new`만 `overload`할 수 있습니다.

**Reply by Lcat12345** [reply]
new로만 예시를 들어서 헷갈리고 있는걸까요

우리가 쓰는 + operator가 new operator와 같은 맥락이고

오버로딩한 operator +가 operator new라고 생각해도 되는건가요?

**Reply by Woon-2** [reply]
보통은 `a + b`와 `operator+(a, b)`가 동치인 것처럼,   
어떤 `operator` `X`가 있다고 할 때   
`a X b`는 `operatorX(a, b)`와 동치입니다.   

그런데 `operator new`는 그렇지 않지요.   
흠...
참고했던 문서를 더 살펴보니

>as to a different phrasing, the standard uses new expression all around, and only once it uses new operator to refer to the place of call. I tend to do the same: operator new for the allocator and new expression for the use.   
> \- [\[Stack Overflow\] Difference between 'new operator' and 'operator new'?](https://stackoverflow.com/questions/1885849/difference-between-new-operator-and-operator-new)

라는 내용을 발견했습니다.

`cppreference`에서도 그렇고, `new operator`보단 `new expression`이 표준 표현인 듯 싶네요.   

마침 해당 과제의 설명력을 좀 높여야겠다 생각했으니    
`new operator`라는 비표준 표현도 `new expression`으로 바꿔야겠습니다.

<br>

그러면 우리가 만드는
```cpp
int* heap_int = new int{100};
```
이라는 `code`는 `new expression`이 적힌 거고,   
`new expression`은 내부적으로 `operator new`를 이용하도록 구현된다고 설명할 수 있겠네요.
